src/workerd/jsg/resource.h

```diff
@@ -1431,7 +1431,7 @@ class ResourceWrapper {
     // We do not allow use of WeakRef or FinalizationRegistry because they introduce
     // non-deterministic behavior.
     check(global->Delete(context, v8StrIntern(isolate, "WeakRef"_kj)));
-    check(global->Delete(context, v8StrIntern(isolate, "FinalizationRegistry"_kj)));
+    //check(global->Delete(context, v8StrIntern(isolate, "FinalizationRegistry"_kj)));
```

Comment:
This probably needs a compat flag as I suspect there are people that check for the existence of these APIs and use them if they are available -- making them suddenly available would therefore cause such workers to start taking a new code path which could end up being broken.

Reply:
This makes me sad, but yes, compat flag absolutely needed. I've seen such checks in the wild a few times now.